### PR TITLE
Mkdocs functions others

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- MKdocs for pivot_longer, pivot_wider, complete, groupby_topk, expand_grid

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by samukweku** [comment]
Apologies team, kindly ignore the DataFrame replacements ... the functions I worked on are just the five listed above(pivot_longer, pivot_wider, groupby_topk, complete, expand_grid)

**Comment by samukweku** [comment]
@loganthomas How do I fix the darglint error?

File: janitor/functions.py [hunk 1]
```diff
@@ -64,27 +64,27 @@
 )
 
 
-def unionize_dataframe_categories(
-    *dataframes, column_names: Optional[Iterable[pd.CategoricalDtype]] = None
+def unionize_DataFrame_categories(
```
Comment (loganthomas) [inline janitor/functions.py:67]: DataFrame doesn't need to be changed in function names or inputs. Please revert this to `unionize_dataframe_cartegories`. We want to follow the Variable and Function naming from [PEP8](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names) here.

File: janitor/functions.py [hunk 2]
```diff
@@ -64,27 +64,27 @@
...
+    *DataFrames, column_names: Optional[Iterable[pd.CategoricalDtype]] = None
```
Comment (loganthomas) [inline janitor/functions.py:68]: Same comment as above. `DataFrame` should remain as `dataframe` in functions, methods, and variables.

File: janitor/functions.py [hunk 3]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
     use:
 
     ```python
-    unionized_dataframes = unionize_dataframe_categories(df1, df2, df2)
-    concatenated_df = pd.concat(unionized_dataframes, ignore_index=True)
+    unionized_DataFrames = unionize_DataFrame_categories(df1, df2, df2)
```
Comment (loganthomas) [inline janitor/functions.py:99]: See comments above

File: janitor/functions.py [hunk 4]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
...
+    concatenated_df = pd.concat(unionized_DataFrames, ignore_index=True)
```
Comment (loganthomas) [inline janitor/functions.py:100]: See comments above

File: janitor/functions.py [hunk 5]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
...
     ```
 
-    :param dataframes: The dataframes you wish to unionize the categorical
+    :param DataFrames: The DataFrames you wish to unionize the categorical
```
Comment (loganthomas) [inline janitor/functions.py:103]: See comments above

File: janitor/functions.py [hunk 6]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
...
     :param column_names: If supplied, only unionize this subset of columns.
-    :returns: A list of the category-unioned dataframes in the same order they
+    :returns: A list of the category-unioned DataFrames in the same order they
         were provided.
     :raises TypeError: if any inputs are not pandas DataFrames.
     """
 
-    if any(not isinstance(df, pd.DataFrame) for df in dataframes):
-        raise TypeError("Inputs must all be dataframes.")
+    if any(not isinstance(df, pd.DataFrame) for df in DataFrames):
```
Comment (loganthomas) [inline janitor/functions.py:111]: See comments above.

File: janitor/functions.py [hunk 7]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
...
+        raise TypeError("Inputs must all be DataFrames.")
```
Comment (loganthomas) [inline janitor/functions.py:112]: See comments above.

File: janitor/functions.py [hunk 8]
```diff
@@ -96,27 +96,27 @@ def unionize_dataframe_categories(
...
 
     if column_names is None:
-        # Find all columns across all dataframes that are categorical
+        # Find all columns across all DataFrames that are categorical
 
         column_names = set()
 
-        for df in dataframes:
+        for df in DataFrames:
```
Comment (loganthomas) [inline janitor/functions.py:119]: See comments above.

**Review by loganthomas — CHANGES_REQUESTED** [review]
@samukweku Thanks for looking to be consistent in the naming convention here! Please update this to only reflect the `dataframe` - > `DataFrame` name change for things that **are not** functions, methods, or variables.

**Comment by loganthomas** [comment]
> @loganthomas How do I fix the darglint error?

@samukweku The best place to start is to look at the error messages of the GitHub action (or if you're running locally, check your pre-commit hooks). I have set up the checks so that they are verbose and will give you a sense of what is failing. For example, from the failing `linting` GitHub action, I see the below:

```
janitor/functions.py:expand_grid:5012: DAR402: Excess exception(s) in Raises section: +r TypeError
```

This means that there is an error on line `5012` in the `expand_grid` function. The error is a darglint `DAR402` error. The message tells me that there is an excess exception in the raises section and that there is an extra `TypeError`. If you look at the source code for `expand_grid()`, there is only a `ValueError` raised.

If you aren't sure what the darglint error is, checking the [dargling docs](https://github.com/terrencepreilly/darglint) is the next best thing to do. I find the [Error Codes](https://github.com/terrencepreilly/darglint#error-codes) section extremely helpful

**Pull request closed without merging**